### PR TITLE
Add opt to use AZ for APISeverLoadBalancer

### PR DESCRIPTION
**Pull request by dulek**
**What this PR does / why we need it**:
Add option to provide AZ for apiserver LB.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [X] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

**Comment by dulek** [comment]
This is fixing #1979.

File: api/v1alpha7/types_conversion.go
Comment (mdbooth) [inline api/v1alpha7/types_conversion.go:519]: Please could you add the new restore function to https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/6c83bc47cf44f8222cedbb7801be0e3e734dec64/api/v1alpha7/conversion_test.go#L375 so it will be fuzzed?

File: api/v1alpha6/types_conversion.go
Comment (mdbooth) [inline api/v1alpha6/types_conversion.go:518]: Please can you add the new restore function to https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/6c83bc47cf44f8222cedbb7801be0e3e734dec64/api/v1alpha6/conversion_test.go#L767 so it will be fuzzed?

**Review by mdbooth — APPROVED** [review]
I've approved, but please add the 2 new restorers to their respective fuzz testers before merging. Thanks!

**Comment by mdbooth** [comment]
Let's get this in.

/hold cancel
/lgtm

**Pull request merged**
